linetools/spectralline.py

```diff
@@ -0,0 +1,370 @@
+            raise ValueError('Rest wavelength must be a Quantity.')
+        self.wrest = wrest
+
+        # Other
+        self.data = {} # Atomic/Moleculare Data (e.g. f-value, A coefficient, Elow)
+        self.analy = {'spec': None, # Analysis inputs (e.g. spectrum; from .clm file or AbsID)
+            'WVMNX': [0., 0.], # Wavelength interval about the line (observed)
+            'VLIM': [0., 0.]*u.km/u.s, # Velocity limit of line, relative to self.attrib['z']
+            'FLG_ANLY': 1 # Analyze
+            }
+        self.attrib = {   # Properties (e.g. column, EW, centroid)
+                       'RA': 0.*u.deg, 'DEC': 0.*u.deg,  #  Coords
+                       'z': 0., 'zsig': 0.,  #  Redshift
+                       'v': 0.*u.km/u.s, 'vsig': 0.*u.km/u.s,  #  Velocity relative to z
+                       'EW': 0.*u.AA, 'EWsig': 0.*u.AA, 'flgEW': 0 # EW
+                       }
+
+        # Fill data
+        self.fill_data(linelist=linelist)
+
+    # Setup spectrum for analysis
+    def cut_spec(self, normalize=False):
+        '''Splice spectrum.  Normalize too (as desired)
+
+        Parameters:
+        ----------
+        normalize: bool, optional
+          Normalize if true (and continuum exists)
+
+        Returns:
+        ----------
+        fx, sig, wave -- Arrays (numpy or Quantity) of flux, error, wavelength
+        '''
+        # Checks
+        if np.sum(self.analy['WVMNX']) == 0.:
+            raise ValueError('spectralline.cut_spec: Need to set WVMNX!') # Could use VMNX
+        if self.analy['spec'] is None:
+            raise ValueError('spectralline.cut_spec: Need to set spectrum!')
+        if self.analy['spec'].wcs.unit == 1.:
+            raise ValueError('Expecting a unit!')
+
+        # Pixels for evaluation
+        pix = self.analy['spec'].pix_minmax(self.analy['WVMNX'])[0]
+
+        # Cut for analysis
+        fx = self.analy['spec'].flux[pix]
+        sig = self.analy['spec'].sig[pix]
+        wave = self.analy['spec'].dispersion[pix]
+
+        # Normalize
+        if normalize:
+            try:
+                fx = fx / self.analy['spec'].conti[pix]
+            except AttributeError:
+                pass
+            else:
+                sig = sig / self.analy['spec'].conti[pix]
+
+        # Return
+        return fx, sig, wave
+
+
+    # EW 
+    def box_ew(self):
+        """  EW calculation
+        Default is simple boxcar integration
+        Observer frame, not rest-frame
+          WVMNX must be set!
+          spec must be set!
+
+        Parameters
+        ----------
+
+        Returns:
+          EW, sigEW : EW and error in observer frame
+        """
+        # Cut spectrum
+        fx, sig, wv = self.cut_spec(normalize=True)
+
+        # dwv
+        dwv = wv - np.roll(wv,1)
+        dwv[0] = dwv[1]
+
+
+        # Simple boxcar
+        EW = np.sum( dwv * (1. - fx) ) 
+        varEW = np.sum( dwv**2 * sig**2 )
+        sigEW = np.sqrt(varEW) 
+
```

Comment:
Are you sure about this one?  

If so, please give me the explicit code.  I fear it isn't
obvious to me.

Thanks.